__tests__/Unit/Pages/Issues.test.tsx

```diff
@@ -0,0 +1,100 @@
+import { waitFor, fireEvent } from '@testing-library/react';
+import Issues from '@/pages/issues';
+import { store } from '@/app/store';
+import { Provider } from 'react-redux';
+import { renderWithRouter } from '@/test_utils/createMockRouter';
+import { setupServer } from 'msw/node';
+import handlers from '../../../__mocks__/handlers';
+import { issuesNoDataFoundHandler } from '../../../__mocks__/handlers/issues.handler';
+
+const server = setupServer(...handlers);
+
+beforeAll(() => {
+    server.listen({ onUnhandledRequest: 'error' });
+});
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
+
+describe('Issues Page', () => {
+    const inputPlaceholderText = 'Enter query string to search issues';
+    const buttonAttribute = {
+        name: 'Submit',
+    };
+    const loadingText = 'Loading...';
+
+    it('Should render search fields in issues page properly', async () => {
+        const { getByPlaceholderText, getByRole, getByText } = renderWithRouter(
+            <Provider store={store()}>
+                <Issues />
+            </Provider>
+        );
+
+        const inputElement = getByPlaceholderText(inputPlaceholderText);
```

Comment:
don't get elements by any kind of text while writing tests

Reply:
I would like to disagree as majority of the APIs from react-testing-library has some version of `getByText` which returns an element that contains the text. I've also seen these these APIs used in production as well.

If there's any credible resource you can cite for this then I'd like to know. Please do share.

Reply:
No, I checked I was wrong here. Somewhere its recommended byTestId, and somewhere byRole, byText etc